src/libAtomVM/interop.h

```diff
@@ -95,14 +96,27 @@ NO_DISCARD enum UnicodeConversionResult interop_chardata_to_bytes(term t, uint8_
  * @details Allows to quickly translate atoms to any integer constant. This function is useful for
  * creating switch statements for atom values.
  * A linear search is performed, so table entries should be sorted by frequency.
- * @param table an array AtomStringIntPair structs, teminated with a default entry marked with
+ * @param table an array AtomStringIntPair structs, terminated with a default entry marked with
  * SELECT_INT_DEFAULT macro.
  * @param atom the atom used for comparison.
  * @param global the global context.
  * @returns the found int value which corresponds to the given atom.
  */
 int interop_atom_term_select_int(const AtomStringIntPair *table, term atom, GlobalContext *global);
 
+/**
+ * @brief Finds the first matching atom in an atoms table .
+ *
+ * @details Allows to quickly translate integer constants to an atom in an atoms table.
+ * This function is the inverse of interop_atom_term_select_int.
+ * @param table an array AtomStringIntPair structs, terminated with a default entry marked with
+ * SELECT_INT_DEFAULT macro.
+ * @param value the in value used for comparison.
+ * @param global the global context.
+ * @returns the found atom which corresponds to the given int value.
+ */
+term interop_atom_term_select_atom(const AtomStringIntPair *table, int value, GlobalContext *global);
```

Comment:
If we want to keep this function we must clearly state that it should be used only when a table is already used, and in any other case a `switch` block should be preferred.

Reply:
Please have a look at the wording I added to the doc comment.